src/utils/make-build-info.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+from datetime import datetime
+from hashlib import md5
+from string import ascii_uppercase
+import json
+import logging
+import os
+
+import requests
+
+
+HEADER_FILE_TEMPLATE = """{license_header}
+
+{generated_warning}
+
+#pragma once
+
+#include "common/streambuf.h"
+
+{defines}
+
+void sbufWriteBuildInfoFlags(sbuf_t *dst);
+"""
+
+SOURCE_FILE_TEMPLATE = """{license_header}
+
+{generated_warning}
+
+#include <stdint.h>
+
+#include "platform.h"
+
+#include "common/streambuf.h"
+
+#include "msp/msp_build_info.h"
+
+void sbufWriteBuildInfoFlags(sbuf_t *dst) 
+{
+    static const uint16_t options[] = {
+{build_options}
+    };
+
+    for (unsigned i = 0; i < ARRAYLEN(options); i++)
+    {
+        sbufWriteU16(dst, options[i]);
+    }
+}
+"""
+
+WARNING_COMMENT_TEMPLATE = """
+/*
+ * WARNING: This is an auto-generated file, please do not edit directly!
+ *
+ * Generator    : `src/utils/make-build-info.py`
+ * Source       : {source}
+ * Input hash   : {input_hash}
+ */
+"""
+
+
+def __find_project_root() -> str:
+    utils_dir = os.path.abspath(os.path.dirname(__file__))
+    src_dir = os.path.dirname(utils_dir)
+    root_dir = os.path.dirname(src_dir)
+    return os.path.realpath(root_dir)
+
+
+def camel_case_to_title(s: str) -> str:
+    if not s:
+        return "Unspecified"
+    else:
+        return "".join([(c if c not in ascii_uppercase else f" {c}") for c in s]).title()
```

Comment:
(just for completeness - this code will not work correctly for input starting with capital letter or containing capital letter after space. ` re.sub(r'\B(?=[A-Z])', ' ', s).title()` is a bit closer but still fails ('toID') )

Reply:
Yes you are right, I just pushed a modified version. Anyway, the serialized data is typically generated from actual objects having member variables and space is not allowed in variable names, it's a valid concern, but a weird edge-case :)